python/pyspark/sql/connect/dataframe.py

```diff
@@ -122,6 +122,20 @@ def withPlan(cls, plan: plan.LogicalPlan, session: "RemoteSparkSession") -> "Dat
         new_frame._plan = plan
         return new_frame
 
+    def isEmpty(self) -> bool:
+        """Returns ``True`` if this :class:`DataFrame` is empty.
+
+        .. versionadded:: 3.4.0
+
+        Returns
+        -------
+        bool
+            Whether it's empty DataFrame or not.
+        """
+        if "is_empty" not in self._cache:
+            self._cache["is_empty"] = len(self.take(1)) == 0
+        return bool(self._cache["is_empty"])
```

Comment:
I think we talked about that do not do cache now but thinking about build general caching layer cc @HyukjinKwon

Reply:
here https://github.com/apache/spark/pull/38546#discussion_r1017482919

Reply:
oh, let me update it. Maybe it is time to design how to do the caching, I will work on it.